config.go

```diff
@@ -34,56 +55,135 @@ func GetCfgDirLocation(baseDir string) (string, error) {
 	return filepath.Join(baseDir, ".ralph-cli"), nil
 }
 
-// PrepareCfgDir creates config dir given as cfgDir (for most cases it will be ~/.ralph-cli)
-// along with its subdirs, and copies bundled scripts to the "scripts" subdir.
-func PrepareCfgDir(cfgDir string) error {
+// PrepareCfgDir creates config dir given as cfgDir (for most cases it will ber
+// ~/.ralph-cli). It also creates default config file, and copies bundled scripts
+// to the scripts subdir.
+func PrepareCfgDir(cfgDir, cfgFileName string) error {
 	var err error
-	err = createCfgDir(cfgDir)
-	if err != nil {
+	if err = createCfgDir(cfgDir); err != nil {
 		return err
 	}
-	var scriptsDir = filepath.Join(cfgDir, "scripts")
+
+	// Create default config file.
+	var cfgFile = filepath.Join(cfgDir, cfgFileName)
+	if !cfgFileExists(cfgFile) {
+		buf := new(bytes.Buffer)
+		if err := toml.NewEncoder(buf).Encode(DefaultCfg); err != nil {
+			return err
+		}
+		if err := ioutil.WriteFile(cfgFile, buf.Bytes(), os.FileMode(0600)); err != nil {
+			return err
+		}
+	}
+
 	// Copy bundled scripts.
+	var scriptsDir = filepath.Join(cfgDir, "scripts")
 	for _, script := range bundledScripts {
 		if _, err := os.Stat(filepath.Join(scriptsDir, script)); os.IsNotExist(err) {
-			err = RestoreAsset(scriptsDir, script)
-			if err != nil {
+			if err = RestoreAsset(scriptsDir, script); err != nil {
 				return err
 			}
 		}
 	}
 	return nil
 }
 
-// createCfgDir is a helper function for creating ~/.ralph-cli dir (used by
-// PrepareCfgDir).
+// createCfgDir is a helper function for creating ~/.ralph-cli dir (used by PrepareCfgDir).
 func createCfgDir(loc string) error {
 	var err error
 	_, err = os.Stat(loc)
 	if os.IsNotExist(err) {
-		mode := os.FileMode(int(0755))
-		err = os.MkdirAll(filepath.Join(loc, "scripts"), mode)
+		err = os.MkdirAll(filepath.Join(loc, "scripts"), os.FileMode(0755))
 		// Add other subdirs to be created here if needed.
 	}
 	return err
 }
 
-// GetConfig loads ralph-cli configuration from ~/.ralph-cli dir.
-func GetConfig() (Config, error) {
-	// TODO(xor-xor): Implement this.
-	return Config{
-		Debug:     false,
-		LogOutput: "",
-	}, nil
+// GetConfig loads ralph-cli configuration from cfgDir (in most cases, it will be
+// ~/.ralph-cli), performs basic validation on it and supplements some of the missing
+// values with their defaults.
+func GetConfig(cfgFile string) (*Config, error) {
+	cfg, err := readConfig(cfgFile)
+	if err != nil {
+		return nil, err
+	}
+	err = cfg.validate()
+	if err != nil {
+		return nil, err
+	}
+	cfg.getDefaults()
```

Comment:
what about environment variables? could we support them? For example we'll read config from toml file, then apply defaults, and then apply env variables on top of that (they will have priority over config file). Then we could specify env variables in during ralph-cli calling, ex. `MANAGEMENT_USER_NAME=admin MANAGEMENT_USER_PASSWORD=secret ralph-cli scan ...` @xor-xor what do you think about it?